crates/volta-core/src/run_package_global/parser.rs

```diff
@@ -0,0 +1,499 @@
+            }
+            _ => CommandArg::NotGlobal,
+        }
+    }
+
+    /// Parse the given set of arguments to see if they correspond to a Yarn global command
+    pub fn for_yarn<S>(args: &'a [S]) -> Self
+    where
+        S: AsRef<OsStr>,
+    {
+        let mut positionals = args.iter().filter(is_positional).map(AsRef::as_ref);
+
+        // Yarn globals must always start with `global <command>`
+        // If we have a global add or remove, then all of the remaining positional arguments will
+        // be the tools to install or uninstall
+        match (positionals.next(), positionals.next()) {
+            (Some(global), Some(add)) if global == "global" && add == "add" => {
+                // The common args for an install should be `global add` and any flags
+                let mut common_args = vec![global, add];
+                common_args.extend(args.iter().filter(is_flag).map(AsRef::as_ref));
+
+                CommandArg::Global(GlobalCommand::Install(InstallArgs {
+                    manager: PackageManager::Yarn,
+                    common_args,
+                    tools: positionals.collect(),
+                }))
+            }
+            (Some(global), Some(remove)) if global == "global" && remove == "remove" => {
+                CommandArg::Global(GlobalCommand::Uninstall(UninstallArgs {
+                    tools: positionals.collect(),
+                }))
+            }
+            _ => CommandArg::NotGlobal,
+        }
+    }
+}
+
+pub enum GlobalCommand<'a> {
+    Install(InstallArgs<'a>),
+    Uninstall(UninstallArgs<'a>),
+}
+
+impl<'a> GlobalCommand<'a> {
+    pub fn executor(self, default_platform: &PlatformSpec) -> Fallible<Executor> {
+        match self {
+            GlobalCommand::Install(cmd) => cmd.executor(default_platform),
+            GlobalCommand::Uninstall(cmd) => cmd.executor(),
+        }
+    }
+}
+
+/// The arguments passed to a global install command
+pub struct InstallArgs<'a> {
+    /// The package manager being used
+    manager: PackageManager,
+    /// Common arguments that apply to each tool (e.g. flags)
+    common_args: Vec<&'a OsStr>,
+    /// The individual tool arguments
+    tools: Vec<&'a OsStr>,
+}
+
+impl<'a> InstallArgs<'a> {
+    /// Convert these global install arguments into an executor for the command
+    ///
+    /// If there are multiple packages specified to install, then they will be broken out into
+    /// individual commands and run separately. That allows us to keep Volta's sandboxing for each
+    /// package while still supporting the ability to install multiple packages at once.
+    pub fn executor(self, default_platform: &PlatformSpec) -> Fallible<Executor> {
+        let mut executors = Vec::with_capacity(self.tools.len());
+
+        for tool in self.tools {
+            match Spec::try_from_str(&tool.to_string_lossy())? {
+                Spec::Package(name, _) => {
+                    let platform = default_platform.as_default();
```

Comment:
This is a little thing that we don't need to tackle now, but needing to do this seems like a signal that our handling of platforms can still keep improving such that we don't end up writing something like this. I wonder if maybe just naming it `platform` would help, so we end up calling `platform.as_default()` instead?